### PR TITLE
Whip Reach Fix

### PR DESCRIPTION
**Pull request by Dariensg**
-Made whip only have extra reach on charge attack
-Added parameter to whip to allow for easy changing of the whip, could be hardcoded for more security but I don't see that being an issue
-Removed the UUID for the reach modifier
-Added ability back to break blocks with whip now that the range is back to normal limitations

### CODE REVIEW

File: src/main/java/com/nekomaster1000/infernalexp/items/IWhipItem.java [hunk 1]
```diff
@@ -1,13 +1,6 @@
 package com.nekomaster1000.infernalexp.items;
```
Comment (caelwarner) [inline src/main/java/com/nekomaster1000/infernalexp/items/IWhipItem.java]: Do we still actually need this interface? Could we possibly remove it as I don't think it really serves any purpose now

**Review by caelwarner — CHANGES_REQUESTED** [review]
Just the one question

File: src/main/java/com/nekomaster1000/infernalexp/items/IWhipItem.java (hunk 1 above)
Comment (Hellion-Games) [reply src/main/java/com/nekomaster1000/infernalexp/items/IWhipItem.java]: That's a good point. I will commit that in a sec

**Review by caelwarner — DISMISSED** [review]
👍

**Review by SwanX1 — CHANGES_REQUESTED** [review]
I don't think that this approach is at all safe. the client determines the reach distance when sending the packet, which can be unsafe.

**Comment by Dariensg** [comment]
We had the same concern about the knockback but decided to leave it as is. I can hard code it on both the client and server side easily, just makes it less convenient to change the reach later. Should I make that change?

**Comment by SwanX1** [comment]
> We had the same concern about the knockback but decided to leave it as is. I can hard code it on both the client and server side easily, just makes it less convenient to change the reach later. Should I make that change?

yes.

**Review by SwanX1 — APPROVED** [review]
LGTM,
Would've preferred if you removed the knockback too,
Merge at your own discretion.

**Review by SwanX1 — APPROVED** [review]

**Pull request merged**